Lib/http/server.py

```diff
@@ -300,6 +300,9 @@ def parse_request(self):
                 #   - Leading zeros MUST be ignored by recipients.
                 if len(version_number) != 2:
                     raise ValueError
+                for component in version_number:
+                    if not component.isdigit():
+                        raise ValueError
```

Comment:
Will this work? I have no Python machine on hands to check myself.

```suggestion
                if any(not component.isdigit() for component in version_number):
                    raise ValueError
```

Reply:
lgtm